realm/realm-library/src/main/java/io/realm/rx/CollectionChange.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2009 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.rx;
+
+import javax.annotation.Nullable;
+
+import io.realm.OrderedCollectionChangeSet;
+import io.realm.OrderedRealmCollection;
+import io.realm.RealmList;
+import io.realm.RealmResults;
+
+/**
+ * Container wrapping the result of a {@link io.realm.OrderedRealmCollectionChangeListener} being triggered.
+ * <p>
+ * This is used by {@link RealmResults#asChangesetObservable()}} and {@link RealmList#asChangesetObservable()} as
+ * RxJava is only capable of emitting one item, not multiple.
+ */
+public class CollectionChange<E extends OrderedRealmCollection> {
```

Comment:
Is `OrderedRealmCollection` raw type on purpose?

Reply:
Not really, but it doesn't matter. When you autocomplete the `new Consumer(...)` it will use the full generic argument.